usr/lib/tik/lib/cenity

```diff
@@ -0,0 +1,258 @@
+cenity() {
+
+  trap ctrl_c INT
+  
+  title=""
+  text=""
+  function=""
+  cancel=true
+  pulsate=false
+  retvalue=0
+  columns=()
+  content=()
+  column_num=0
+
+  for p in "$@"; do
+    case "$p" in
+      --info*)
+        function=c_info
+        ;;
+      --warning*)
+        function=c_warning
+        ;;
+      --error*)
+        function=c_error
+        ;;
+      --question*)
+        function=c_question
+        ;;
+      --entry*)
+        function=c_entry
+        ;;
+      --password*)
+        function=c_password
+        ;;
+      --list*)
+        function=c_list
+        ;;
+      --progress*)
+        function=c_progress
+        ;;
+      --title*)
+        title=$(sed 's/^--title=//' <<< $p)
+        ;;
+      --text*)
+        text=$(sed 's/^--text=//' <<< $p)
+        ;;
+      --percentage*)
+        percentage=$(sed 's/^--percentage=//' <<< $p)
+        ;;
+      --column*)
+        columns+=("$(cut -d '=' -f2 <<< $p)")
+        ;;
+      --pulsate*)
+        pulsate=true
+        ;;
+      --no-cancel*)
+        cancel=false
+        ;;
+      "")
+        ;;
+      *)
+        content+=("$p")
+        ;;
+    esac
+  done
+
+  ctrl_c() {
+    if ! $cancel; then
+      echo "No Cancel"
+    else
+      echo "Canceled"
+      retvalue=1
+      return $retvalue
+    fi
+  }
+
+  if [[ ! $function ]]; then
+    echo "function not set"
+    return 1
+  fi
+
+  text=$(c_replace_tags "$text")
+
+  if [ $function = "c_progress" ]; then
+    echo -e "--PROGESS--- ${title} -------------\n"
+    if $cancel; then
+      echo "Progress (CTRL + C to cancel): "
+    fi
+
+    if [ -p /dev/stdin ]; then
+      if $pulsate; then
+        cat -
+      else
+        while IFS="" read line; do
+          c_progress "${line}"
+        done
+      fi
+    fi
+  else
+    result=""
+    $function result "$title" "$text" columns content; retvalue=$?
+  fi
+
+  eval $1='$result'
+  return $retvalue
+}
+
+
+c_clear() {
+  printf "\033c"
+}
+
+
+c_replace_tags() {
+  text="$1"
+  text="$(echo "$text" | sed -e 's/\(<b>\|<tt>\|<big>\)/\\033[1m/g' | sed 's/\(<\/b>\|<\/tt>\|<\/big>\)/\\033[0m/g')"
+  text="$(echo "$text" | sed -e 's/<span face=.*><\/span>//')"
+  echo "$text"
+}
+
+
+c_key() {
+  echo -e "\tPress key to continue"
+  read
```

Comment:
`read -n 1 -s -r` would be better here, plain `read` lets you type junk and needs an Enter to continue